cli/internal/codegen/golang.go

```diff
@@ -8,18 +8,25 @@ import (
 
 	"github.com/cockroachdb/errors"
 	. "github.com/dave/jennifer/jen"
+	"github.com/fatih/structtag"
 
 	"encr.dev/cli/internal/version"
+	"encr.dev/internal/gocodegen"
+	"encr.dev/parser/encoding"
 	meta "encr.dev/proto/encore/parser/meta/v1"
 	schema "encr.dev/proto/encore/parser/schema/v1"
 )
 
 type golang struct {
-	md *meta.Data
+	md  *meta.Data
+	enc *gocodegen.MarshallingCodeGenerator
 }
 
 func (g *golang) Generate(buf *bytes.Buffer, appSlug string, md *meta.Data) (err error) {
 	g.md = md
+	// note we use unicode characters to write "encore" to ensure we don't conflict with any existing names in the
+	// package the generated client is placed.
+	g.enc = gocodegen.NewMarshallingCodeGenerator("ℯ𝓃𝑐ℴ𝑟ℯMarshaller")
```

Comment:
Nit: I'm not a big fan of unicode chars in variable names, but I guess it's just a matter of preferences :)